### PR TITLE
BZ2056385: Adds mirror registry example

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.9+

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2056385

Link to docs preview: http://file.rdu.redhat.com/sdudhgao/mirrir-reg/openshift_images/image-configuration.html#images-configuration-registry-mirror_image-configuration

Additional information:

### CODE REVIEW

File: modules/images-configuration-registry-mirror.adoc [hunk 1]
```diff
@@ -84,10 +84,15 @@ spec:
...
     source: registry.redhat.io/openshift4 <3>
+  - mirrors:
+    - mirror.example.com <4>
+    source: registry.redhat.io <5>
 ----
 <1> Indicates the name of the image registry and repository.
 <2> Indicates the registry and repository containing the content that is mirrored.
 <3> You can configure a namespace inside a registry to use any image in that namespace. If you use a registry domain as a source, the `ImageContentSourcePolicy` resource is applied to all repositories from the registry.
+<4> Indicates the name of the mirror.
+<5> Indicates the name of the registry.
```
Comment (xiuwang) [inline modules/images-configuration-registry-mirror.adoc:95]: Could we add those content like in https://bugzilla.redhat.com/show_bug.cgi?id=2056385#c2 ?
```
source: registry.example.com/foo/bar/myapp
mirrors: mirror.example.net/app
-> result: mirror.example.net/app@sha256:...

source: registry.example.com/foo/bar
mirrors: mirror.example.net
-> result: mirror.example.net/myapp@sha256:...

source: registry.example.com
mirrors: mirror.example.net/registry-example-com
-> result: mirror.example.net/registry-example-com/foo/bar/myapp@sha256:...

source: registry.example.com/foo
mirrors: mirror.example.net/mirrored-foo
-> result: mirror.example.net/mirrored-foo/bar/myapp@sha256:...
```

Comment (xiuwang) [reply modules/images-configuration-registry-mirror.adoc:95]: Hi @xenolinux ,Please update per your understand, we can update registryrepomirror.yaml, such as
```
spec:
  repositoryDigestMirrors:
  - mirrors:
    - example.io/example/ubi-minimal <1>
    - example.com/example/ubi-minimal <2>
    source: registry.access.redhat.com/ubi8/ubi-minimal <3>
  - mirrors:
    - mirror.example.com/redhat
    source: registry.redhat.io/openshift4 <4>
  - mirrors:
    - mirror.example.com
    source: registry.redhat.io <5>
  - mirrors:
    - mirror.example.net/image
    source: registry.example.com/example/myimage <6>
  - mirrors:
    - mirror.example.net
    source: registry.example.com/example <7>
  - mirrors:
    - mirror.example.net/registry-example-com
    source: registry.example.com <8>
```
<2>  Identify multiple mirrored repositories for each target repository, to make sure that if one mirror is down, another can be used
<5> You can only configure the name of registry, then ImageContentSourcePolicy resource is applied to all repositories from source registry to mirror registry, the namespace should be the same.
<6> mirror.example.net/image@sha256:... will be pulled
<7>myimage under example namespace of source registry will be pulled from mirror.example.net/myimage@sha256:...
<8>ImageContentSourcePolicy resource is applied to all repositories of source registry to mirror.example.net/registry-example-com, registry.example.com/example/myimage will be pulled from mirror.example.net/registry-example-com/example/myimage@sha256:...

File: modules/images-configuration-registry-mirror.adoc [hunk 2]
```diff
@@ -77,17 +77,32 @@ spec:
...
+    source: registry.example.com/example <7>
+  - mirrors:
+    - mirror.example.net/registry-example-com
+    source: registry.example.com <8>
 ----
 <1> Indicates the name of the image registry and repository.
-<2> Indicates the registry and repository containing the content that is mirrored.
+<2> Indicates multiple mirror repositories for the each target repository. If a one mirror is down, the target repository can use another mirror.
 <3> You can configure a namespace inside a registry to use any image in that namespace. If you use a registry domain as a source, the `ImageContentSourcePolicy` resource is applied to all repositories from the registry.
+<4> Indicates the name of the mirror.
```
Comment (xiuwang) [inline modules/images-configuration-registry-mirror.adoc:101]: the content of <3> should explain for <4>
<3>Indicates the registry and repository containing the content that is mirrored.

File: modules/images-configuration-registry-mirror.adoc [hunk 3]
```diff
@@ -77,17 +77,32 @@ spec:
...
+<5> If you configure the registry name in that namespace, the `ImageContentSourcePolicy` resource is applied to all repositories from a source registry to a mirror registry.
```
Comment (xiuwang) [inline modules/images-configuration-registry-mirror.adoc:102]: How about s/If you configure the registry name in that namespace/If you only configure the registry name, ?

The namespace of mirror registry will keep same with source registry.

File: modules/images-configuration-registry-mirror.adoc [hunk 4]
```diff
@@ -77,17 +77,32 @@ spec:
...
+<6> Pulls the image `mirror.example.net/image@sha256:...`.
+<7> Pulls the image `myimage` in the source registry namespace from the source mirror `mirror.example.net/myimage@sha256:...`.
```
Comment (xiuwang) [inline modules/images-configuration-registry-mirror.adoc:104]: s/source mirror/mirror

File: modules/images-configuration-registry-mirror.adoc [hunk 5]
```diff
@@ -77,17 +77,32 @@ spec:
...
+<8> Pulls the image `registry.example.com/example/myimage` from the source registry `mirror.example.net/registry-example-com/example/myimage@sha256:...`. The `ImageContentSourcePolicy` resource is applied to all repositories from a source registry to a mirror registry `mirror.example.net/registry-example-com`.
```
Comment (xiuwang) [inline modules/images-configuration-registry-mirror.adoc:105]: s/the source registry/ the mirror registry

File: modules/images-configuration-registry-mirror.adoc (hunk 3 above)
Comment (xenolinux) [reply modules/images-configuration-registry-mirror.adoc]: Addressed

File: modules/images-configuration-registry-mirror.adoc (hunk 4 above)
Comment (xenolinux) [reply modules/images-configuration-registry-mirror.adoc:104]: Addressed

File: modules/images-configuration-registry-mirror.adoc (hunk 5 above)
Comment (xenolinux) [reply modules/images-configuration-registry-mirror.adoc:105]: Addressed

File: modules/images-configuration-registry-mirror.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/images-configuration-registry-mirror.adoc:101]: Addressed

Comment (xiuwang) [reply modules/images-configuration-registry-mirror.adoc:101]: Could we update <4> ?
<4> You can configure a namespace inside a registry to use any image in that namespace. If you use a registry domain as a source, the `ImageContentSourcePolicy` resource is applied to all repositories from the registry.

**Comment by xenolinux** [comment]
closing this PR in favor of https://github.com/openshift/openshift-docs/pull/48766

File: modules/images-configuration-registry-mirror.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/images-configuration-registry-mirror.adoc:101]: Acked

**Pull request closed without merging**